ansible/roles/operator-pipeline/templates/openshift/pipelines/community-release-pipeline.yml

```diff
@@ -173,17 +174,47 @@ spec:
           workspace: repository
           subPath: base
 
+    # acquire/lease the resource to resolve the conflict of concurrent pipelineruns
+    - name: acquire-lease
```

Comment:
Is it really necessary to acquire a lease at the beginning of the pipeline? Is there any other critical section other than IIB build?

Reply:
I moved it here just to ensure that the release of the lease in the `finally` section has a matching acquire task even when we skip the actual release steps, but I'm not sure it's required. And the tasks between this and the start of the actual critical section are just a couple and quite simple.

Reply:
The delete lease is implemented like `oc delete lease $LEASE_NAME --ignore-not-found=true` so I think it should not be a problem. My concern about moving the acquire task to the early stages of the pipeline is that we already faced an issue in ISV release pipeline when the lease caused a bottleneck when multiple PRs have been merged simultaneously. We should aim to have the critical section as small as possible.

Reply:
My concern is that when we get to the `oc delete` step without acquiring the lease first, it could delete the lease acquired by another pipeline run: I don't see any mechanism in place to avoid that.
Anyway, I've moved the task back to its original position and added a `when` clause to the lease release task but still need to iron out a remaining issue. Will ping you when it's ready for another look.